tests/PHPStan/Analyser/nsrt/static-late-binding.php

```diff
@@ -112,6 +112,15 @@ public function foo(): void
         assertType('bool', $v);
         $clUnioned->outStaticConst($v);
         assertType('bool', $v); // should be bool|int
+
+        assertType('StaticLateBinding\A', A::retStatic(...)());
+        assertType('StaticLateBinding\B', B::retStatic(...)());
+        assertType('Closure', self::retStatic()(...)); // should be static(StaticLateBinding\B)
```

Comment:
This shows this doesn't really work well.

Reply:
This PR is fine - the issue is unrelated - https://phpstan.org/r/d05a08bd-650e-4e96-8a2f-e400ed74dcb0

Reply:
@ondrejmirtes can this PR be merged?